libraries/commerce/product/streams/index.js

```diff
@@ -63,6 +63,9 @@ export const receivedVisibleProduct$ = productReceived$.merge(cachedProductRecei
     return action.productData.id === hex2bin(route.params.productId);
   });
 
+export const receivedVisibleCachedProduct$ = receivedVisibleProduct$
+  .filter(({ action }) => action.type === RECEIVE_PRODUCT_CACHED);
```

Comment:
This stream emits whenever `RECEIVE_PRODUCT_CACHED` is dispatched - independent from opening the page. So when you have a product with variants and switch between them, a getProduct request is dispatched again for any previously selected variant. So the caching is basically deactivated.